beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deneb.go

```diff
@@ -1,12 +1,59 @@
 package validator
 
 import (
+	"sync"
+
 	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
 	enginev1 "github.com/prysmaticlabs/prysm/v4/proto/engine/v1"
 	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/v4/runtime/version"
 )
 
+var bundleCache = &blobsBundleCache{
+	blobs: make(map[primitives.Slot]*enginev1.BlobsBundle),
+}
+
+// BlobsBundleCache holds the KZG commitments and other relevant sidecar data for a local beacon block.
+type blobsBundleCache struct {
+	blobs map[primitives.Slot]*enginev1.BlobsBundle
+	sync.Mutex
+}
+
+// add adds a blobs bundle to the cache.
+// same slot overwrites the previous bundle.
+func (c *blobsBundleCache) add(slot primitives.Slot, bundle *enginev1.BlobsBundle) {
+	c.Lock()
+	c.blobs[slot] = bundle
+	c.Unlock()
+
+	// Trigger pruning in the background
+	go c.prune(slot)
+}
+
+// get gets a blobs bundle from the cache.
+func (c *blobsBundleCache) get(slot primitives.Slot) *enginev1.BlobsBundle {
+	c.Lock()
+	blobs := c.blobs[slot]
+	c.Unlock()
+
+	// Trigger pruning in the background
+	go c.prune(slot)
```

Comment:
Ignoring my other comment, if you want the cache to be some small size larger than 1 then I would assume spawning a goroutine isn't worth it and you could just prune inline.

Reply:
that was my original intent, but Potuz pointed out that several use cases may cause memory leaks in the long term
- caller like vouch may call `GetBlock` without `ProposeBlock`
- caller may call `GetBlock` once

without pruning routine, the worst case is we'll always keep a blobs bundle in the memory

Reply:
I'm confused. I'm not suggesting we don't prune, I'm saying that, since we prune on add and retrieve, there's no chance of prune not being called and the map growing to a large size, so it's always so cheap to prune it that it doesn't make sense to spawn a goroutine.